### DISCUSSION TITLE
Depth resolution in iw3 is broken past 518

### DISCUSSION THREAD

**Discussion by kaelsonofkrypto** [nagadomi/nunif#150] [General]
@nagadomi Hi Nagadomi, Thanks to a recent discovery by another member, we found that depth resolution for any S v2 degrades in quality when it exceeds 518 depth resolution, resulting in a worse appearance. However, using any S v2 at a 518 depth resolution produces much better results compared to higher depths. The depth resolution in IW3 appears to be broken beyond 518 depth resolutio as demonstrated in the attached screenshot where IW3 any S v2 at a depth resolution of 1400 looks significantly worse than when set to 518 depth resolution. In theory, increasing the depth resolution should enhance the quality, but unfortunately, this isn't happening. Could you please examine and fix the depth resolution field for values beyond 518 depth resolution? Its broke across the other models too, i did the test on any large as well. you can see the higher depth res on any large also looks worse than the any large at 518 in my screenshots below

![image](https://github.com/nagadomi/nunif/assets/164638883/f938ea00-36bf-4fa8-9d4a-41e9129fccf7)

![image](https://github.com/nagadomi/nunif/assets/164638883/2bf84cc1-8d6c-4944-895a-74599a336e20)

**Comment by nagadomi** [comment]
See my comment in #141 it is the same problem.

I tried V2 Small Depth Resolution=1000 with the official code and get results similar to the left. (1400 could not be executed due to  out of memory)
So I think this is a model problem, not iw3.

---
I think your group could use `run.py` in the official repository for a detailed comparison. It can change the depth resolution (--input-size).
example command
```
python run.py --img-path test2.png --pred-only --grayscale --encoder vits --input-size 1000
```
https://github.com/DepthAnything/Depth-Anything-V2?#running-script-on-images

**Reply by kaelsonofkrypto** [reply]
thanks nagadomi, we will report the findings to the any devs, but if u dont mind, since u are a dev and more knowledgeable on this than us, can u also report the findings of how depth maps behave with the models  to the any devs on this, so they can tweak it to work properly with high depth resolution. thank you so much for your insights, as i think those high depth resolutions are important, its how we can close the gap to get movie theater 3d

**Reply by nagadomi** [reply]
I think of DepthAnything as research and not software development, so I am not going to complain about the results.
Honestly, I think it is unlikely that requests for improvements not described in the paper will be approved.

**Reply by kaelsonofkrypto** [reply]
thanks for all your help either way, appreciate you
